src/Controller/Controller.php

```diff
@@ -572,6 +572,9 @@ public function render($view = null, $layout = null)
             return $this->response;
         }
 
+        // Re-fetch View class to pass view variables set in beforeRender callbacks
+        $this->View = $this->getView();
+
         if ($viewClass !== $this->viewClass) {
             $this->View = $this->getView($this->viewClass);
```

Comment:
I like the idea of a proxy as well. We can merge this, and then add the proxy afterwards. We would need to update some of the workflow methods as getView() should probably continue to return a view instance or we'll have BC issues. However, there could be other methods that act on the view configuration data before the view is constructed.

How does something like

```php
$this->viewConfig('layout', 'admin');
// or
$this->view()->layout('admin');
```
feel to people?